sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQuerySchemaIOProvider.java

```diff
@@ -17,199 +17,63 @@
-   *       created if it does not exist. Valid inputs are "Never" and "IfNeeded", corresponding to
-   *       values of {@link BigQueryIO.Write.CreateDisposition}. Used as an input to {@link
-   *       BigQueryIO.Write#withCreateDisposition(BigQueryIO.Write.CreateDisposition)}.
-   * </ul>
-   *
-   * Relevant default values for these transforms that are not configurable fields are as follows:
-   *
-   * <ul>
-   *   <li>ReadMethod - The input to {@link
-   *       BigQueryIO.TypedRead#withMethod(BigQueryIO.TypedRead.Method)}. Defaults to EXPORT, since
-   *       that is the only method that currently offers Beam Schema support.
-   *   <li>WriteMethod - The input to {@link BigQueryIO.Write#withMethod(BigQueryIO.Write.Method)}.
-   *       Currently defaults to STORAGE_WRITE_API.
-   * </ul>
-   */
+  /** Returns the expected class of the configuration object. */
   @Override
-  public Schema configurationSchema() {
-    return Schema.builder()
-        .addNullableField("table", FieldType.STRING)
-        .addNullableField("query", FieldType.STRING)
-        .addNullableField("queryLocation", FieldType.STRING)
-        .addNullableField("createDisposition", FieldType.STRING)
-        .build();
-  }
-
-  private static final HashMap<String, BigQueryIO.Write.CreateDisposition> createDispositionsMap =
-      new HashMap<>();
-
-  static {
-    createDispositionsMap.put("Never", BigQueryIO.Write.CreateDisposition.CREATE_NEVER);
-    createDispositionsMap.put("IfNeeded", BigQueryIO.Write.CreateDisposition.CREATE_IF_NEEDED);
+  public Class<BigQuerySchemaIOConfiguration> configurationClass() {
+    return BigQuerySchemaIOConfiguration.class;
   }
 
   /**
-   * Produces a SchemaIO given a String representing the data's location, the schema of the data
-   * that resides there, and some IO-specific configuration object.
-   *
-   * <p>For BigQuery IO, only the configuration object is used. Location and data schema have no
-   * effect. Specifying a table and dataset is done through appropriate fields in the configuration
-   * object, and the data schema is automatically generated from either the input PCollection or
-   * schema of the BigQuery table.
+   * Produces a {@link SchemaTransform} implementation based on the {@link
+   * BigQuerySchemaIOConfiguration} details.
    */
   @Override
-  public BigQuerySchemaIO from(String location, Row configuration, @Nullable Schema dataSchema) {
-    return new BigQuerySchemaIO(location, configuration);
+  public SchemaTransform from(BigQuerySchemaIOConfiguration configuration) {
+    return BigQuerySchemaTransform.of(configuration);
   }
 
-  /**
-   * Indicates whether this transform requires a specified data schema.
-   *
-   * @return false
-   */
+  /** Returns an id that uniquely identifies this transform. */
   @Override
-  public boolean requiresDataSchema() {
-    return false;
+  public String identifier() {
+    return BigQuerySchemaIOConfiguration.IDENTIFIER;
```

Comment:
Conceptually this doesn't really belong in the configuration class. I'd either do a string or a constant described in this file